url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```diff
@@ -0,0 +1,714 @@
+
+        @ScriptyCommand(name="Integer")
+        public static TypeTuple integerType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return INTEGER_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        @ScriptyCommand(name="BigInteger")
+        public static TypeTuple bigIntegerType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BIGINTEGER_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        @ScriptyCommand(name="Boolean")
+        public static TypeTuple booleanType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BOOLEAN_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        @ScriptyCommand(name="Byte")
+        public static TypeTuple byteType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BYTE_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+        
+        // type="ListOf (Integer) minLength=0 maxLength=20"
+        private static StdArgList LISTOF_ARGLIST = new StdArgList(new FixedArg[]{FIXED_TYPE}, new OptionalArg[]{}, new NamedArg[]{NAMED_MINLEN, NAMED_MAXLEN});
+        
+        @ScriptyCommand(name="ListOf")
+        public static TypeTuple listOfType(IContext aCtx, Object[] aArgs)
+        throws CommandException
+        {
+            try
+            {
+                aArgs = LISTOF_ARGLIST.guard(aArgs, aCtx);
+
+                TypeTuple lTuple = (TypeTuple) aArgs[1];
+                Object lTypeCandidate = lTuple.getX();
+                Map<String, IArgMapping> lMappings = lTuple.getY();
+
+                if(lTypeCandidate == null || !(lTypeCandidate instanceof ITypeSpec))
+                {
+                    throw new CommandException("Invalid type expression ListOf(???) for the elements of the list.");
+                }
+
+                ITypeSpec lSpec = (ITypeSpec) lTypeCandidate;
+                int lMin = (Integer) aArgs[2];
+                int lMax = (Integer) aArgs[3];
+
+                return new TypeTuple(new CheckedListType(lSpec, lMin, lMax), lMappings);
+            }
+            catch (ArgSpecException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)